src/Common/tests/System/Net/RemoteServerQuery.cs

```diff
@@ -0,0 +1,79 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Test.Common
+{
+    internal static class RemoteServerQuery
+    {
+        internal static async Task Run(Func<Task> testCode, Type remoteServerFailExceptionType, string servername, bool mustFail = true)
+        {
+            try
+            {
+                await testCode();
+            }
+            catch (Exception actualException)
+            {
+                if (remoteServerFailExceptionType.Equals(actualException.GetType()))
+                {
+                    if (mustFail)
+                        Assert.True(false, $"External issue with remote server: {servername}");
```

Comment:
I think we should report the original exception. This way we entirely swallow it. Which may not be desirable in case we make a mistake in the test.
What about wrapping the exception in a new exception type "RemoteServerException" which will have this text, serverName field and InnerException?

Reply:
we should start thinking about using xunit output. That would allow us to capture more than just single stack trace. 
I'm also not sure what exactly this is trying to solve. It would be very nice to see what was happening before and how does it look now.   My assumption is that we would like to see distinct pattern for cases where we fail to talk to external service. That should simplify triage but it should be scoped only to IO operations IMHO.